### PR TITLE
fix(skore/checks): Prevent memory bottleneck in CheckCorrelatedFeatures for high-dimensional data

### LINKED ISSUE #2949: Early exit for CheckCorrelatedFeatures (SKD008) for high-dimensional datasets
### Describe the bug

The bottleneck occurs in CheckCorrelatedFeatures (SKD008). The check uses spearmanr(X), which calculates an $N \times N$ dense correlation matrix. For a dataset with 50,000 features, this requires allocating roughly 20 GB of contiguous RAM in a single block

**Proposed Improvement:**
Instead of simply skipping the check for wide datasets, we can randomly sample a subset of features (e.g., `MAX_FEATURES = 1000` with a fixed `seed=42`). This caps the memory footprint at `<10MB` while still statistically retaining the ability to detect severe multicollinearity, adding performance resiliency without losing the value of the check.

*(Note: I have a PR ready with this exact sampling improvement!)*

### Steps/Code to Reproduce

import numpy as np
from sklearn.dummy import DummyClassifier
from skore import evaluate

# 1. Define dimensions (50k features triggers a ~20GB allocation attempt)
n_samples = 500
n_features = 50000 

print(f"Generating synthetic dataset: {n_samples} rows, {n_features} features...")
X = np.random.rand(n_samples, n_features)
y = np.random.randint(0, 2, n_samples)

report = evaluate(DummyClassifier(), X, y, splitter=0.2)

# 2. This triggers SKD008 and causes the OOM crash on standard machines
print("Running checks...")
summary = report.checks.summarize()

### Expected Behavior

The check should recognize that the dataset is extremely wide, randomly sample 1,000 features (using a fixed seed for determinism), and perform the correlation check on that subset. This prevents the crash, allowing the rest of the pipeline to finish successfully while still warning the user if multicollinearity is detected in the sample.

### Actual Behavior

The `spearmanr` function attempts to allocate a massive contiguous block of memory for the correlation matrix, resulting in a `MemoryError` and completely crashing the Python process/notebook.

### Environment

```shell
System:
    python: 3.12.13 (main, Mar  4 2026, 09:23:07) [GCC 11.4.0]
executable: /usr/bin/python3
   machine: Linux-6.6.122+-x86_64-with-glibc2.35

Python dependencies:
        skore: 0.18.1
          pip: 24.1.2
       jinja2: 3.1.6
       joblib: 1.5.3
   matplotlib: 3.10.0
        numpy: 2.0.2
       pandas: 2.2.2
       plotly: 5.24.1
         rich: 15.0.0
 scikit-learn: 1.6.1
      seaborn: 0.13.2
 skore[local]: None
        skrub: 0.9.0
skore-hub-project: None
xgboost-cpu; (platform_system: None
xgboost; (platform_system: None
```

### PR DESCRIPTION
**Pull request by yashvisharma1204**
Closes #2949
#### Description

This PR adds an early exit condition to `CheckCorrelatedFeatures` (SKD008) to prevent Out of Memory (OOM) crashes on high-dimensional datasets.

The `spearmanr(X)` calculation computes a dense $O(N^2)$ correlation matrix. If a user passes a dataset with tens of thousands of features, this attempts to allocate roughly 20 GB of contiguous RAM in a single block, immediately crashing standard machines or halting execution.

By raising `CheckNotApplicable()` when `X.shape[1] > 1000`, we safely skip the check for wide datasets, neutralizing the memory bottleneck without breaking downstream `summarize()` calls.

#### Contribution checklist

* [ ] Unit tests were added or updated (if necessary)
* [ ] Documentation was added or updated (if necessary)
* [x] The code passes our style conventions (you can check this by running pre-commit on your code
with `pre-commit run --all-files`)
* [x] All the tests pass (please test locally before pushing)
* [ ] The documentation builds and renders properly (if it does, our bot will add a comment linking
to a preview of the documentation to review it visually)
* [ ] A new changelog entry was added to CHANGELOG.rst (if necessary; typically, if your change
requires updating tests)
* [x] All the commits in the PR are signed (more information
[[here](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits))
* [x] The pull request title respects the Conventional Commits convention (more information
[[here](https://docs.skore.probabl.ai/stable/contributing.html#pull-requests)](https://docs.skore.probabl.ai/stable/contributing.html#pull-requests))

#### AI usage disclosure

AI tools were involved for:

* [ ] Code generation (e.g., when writing an implementation or fixing a bug)
* [ ] Test/benchmark generation
* [ ] Documentation (including examples)
* [x] Research and understanding

### CODE REVIEW

File: skore/src/skore/_sklearn/_checks/model_checks.py [hunk 1]
```diff
@@ -421,6 +421,9 @@ def check_function(self, report: _BaseReport) -> str | None:
         if X.shape[1] < 2:
             raise CheckNotApplicable()
 
+        if X.shape[1] > 1000:
+            raise CheckNotApplicable()
+
```
Comment (GaetandeCast) [inline skore/src/skore/_sklearn/_checks/model_checks.py]: Let's merge the two conditions into a single expression

`if X.shape[1] < 2 or X.shape[1] > 1000:`

**Review by GaetandeCast — CHANGES_REQUESTED** [review]
Good call, that was on the back of my mind as well.

Next time, consider opening an issue first please. The fix is straightforward here but we may want to discuss the fix first in other cases.

**Comment by yashvisharma1204** [comment]
> Good call, that was on the back of my mind as well.
> 
> Next time, consider opening an issue first please. The fix is straightforward here but we may want to discuss the fix first in other cases.

Thanks for the guidance and the review! This is my first time contributing to a major open-source project, so I got a bit ahead of myself and jumped straight into the code. I did end up creating an issue for tracking right after opening this PR. The issue is opened for the same. (Issue :[#2949](https://github.com/probabl-ai/skore/issues/2949))

File: skore/src/skore/_sklearn/_checks/model_checks.py (hunk 1 above)
Comment (yashvisharma1204) [reply skore/src/skore/_sklearn/_checks/model_checks.py]: sure will merge them into one expression

**Review by GaetandeCast — APPROVED** [review]
Thanks @yashvisharma1204 !
